src/Microsoft.AspNet.Http/Features/FormFeature.cs

```diff
@@ -98,10 +123,10 @@ public IFormCollection ReadForm()
             _request.EnableRewind();
 
             IDictionary<string, StringValues> formFields = null;
-            var files = new FormFileCollection();
+            FormFileCollection files = null;
 
             // Some of these code paths use StreamReader which does not support cancellation tokens.
-            using (cancellationToken.Register(_request.HttpContext.Abort))
+            using (cancellationToken.Register((Action)_request.HttpContext.Abort))
```

Comment:
Why are you casting here? You can actually avoid the delegate allocation per request here as well.

```
cancellationToken.Register(state => ((HttpContext)state).Abort, _request.HttpContext)
```